node-test/src/test/scala/org/bitcoins/node/networking/peer/DataMessageHandlerTest.scala

```diff
@@ -0,0 +1,177 @@
+package org.bitcoins.node.networking.peer
+
+import org.bitcoins.core.currency._
+import org.bitcoins.core.gcs.{FilterType, GolombFilter}
+import org.bitcoins.core.p2p._
+import org.bitcoins.core.protocol.blockchain.{Block, BlockHeader, MerkleBlock}
+import org.bitcoins.core.protocol.transaction.Transaction
+import org.bitcoins.core.protocol.{BitcoinAddress, CompactSizeUInt}
+import org.bitcoins.crypto.DoubleSha256Digest
+import org.bitcoins.node._
+import org.bitcoins.rpc.client.common.BitcoindVersion
+import org.bitcoins.rpc.client.v19.BitcoindV19RpcClient
+import org.bitcoins.server.BitcoinSAppConfig
+import org.bitcoins.testkit.BitcoinSTestAppConfig
+import org.bitcoins.testkit.node.NodeUnitTest
+import org.bitcoins.testkit.node.fixture.SpvNodeConnectedWithBitcoind
+import org.scalatest.FutureOutcome
+
+import scala.concurrent.{Future, Promise}
+
+class DataMessageHandlerTest extends NodeUnitTest {
+
+  /** Wallet config with data directory set to user temp directory */
+  implicit override protected def config: BitcoinSAppConfig =
+    BitcoinSTestAppConfig.getSpvWithEmbeddedDbTestConfig(pgUrl)
+
+  override type FixtureParam = SpvNodeConnectedWithBitcoind
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome =
+    withSpvNodeConnectedToBitcoind(test, Some(BitcoindVersion.V19))
+
+  private val junkAddress: BitcoinAddress =
```

Comment:
how many junk addresses are you creating? lol

Reply:
It's the same one I keep copying over, maybe should move it into the util file xD